landscape/package/tests/test_reporter.py

```diff
@@ -926,6 +933,51 @@ def got_result(result):
         result = self.reporter.detect_packages_changes()
         return result.addCallback(got_result)
 
+    def test_detect_packages_changes_with_autoremovable(self):
+        message_store = self.broker_service.message_store
+        message_store.set_accepted_types(["packages"])
+
+        self.store.set_hash_ids({HASH1: 1, HASH2: 2, HASH3: 3})
+        self.store.add_available([1, 2, 3])
+        self.store.add_installed([1])
+        self.set_pkg1_autoremovable()
+
+        result = self.successResultOf(self.reporter.detect_packages_changes())
+        self.assertTrue(result)
+
+        expected = [{"type": "packages", "autoremovable": [1]}]
+        self.assertMessages(message_store.get_pending_messages(), expected)
+        self.assertEqual([1], self.store.get_autoremovable())
+
+    def test_detect_packages_changes_with_not_autoremovable(self):
+        message_store = self.broker_service.message_store
+        message_store.set_accepted_types(["packages"])
+
+        self.store.set_hash_ids({HASH1: 1, HASH2: 2, HASH3: 3})
+        self.store.add_available([1, 2, 3])
+        self.store.add_autoremovable([1, 2])
```

Comment:
Don't confuse the client store state with the apt package status. Unless the apt package is marked auto and not depended upon, it would still be flagged as not-autoremovable. It's true that in practice the installed state will come with autoremovable state in the store, but I don't care about exercising that path in this test.

I'll add a comment. on the logic